sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -295,6 +296,8 @@ case class PreprocessTableCreation(catalog: SessionCatalog) extends Rule[Logical
         c.copy(tableDesc = normalizedTable, query = Some(reorderedQuery))
       } else {
         DDLUtils.checkTableColumns(tableDesc)
+        SchemaUtils.checkIndeterminateCollationInSchema(tableDesc.schema)
```

Comment:
We don't need to check it here as CREATE TABLE can't produce indeterminate collation, only CTAS can.